python/pyarrow/_fs.pyx

```diff
@@ -541,6 +754,12 @@ cdef class FileSystem(_Weakrefable):
             The path of the file to be copied from.
         dest : str
             The destination path where the file is copied to.
+
+        Examples
+        --------
+        >>> local.copy_file('/tmp/fileinfo.dat', '/tmp/fileinfo_copy.dat')
+        >>> local.get_file_info('/tmp/fileinfo_copy.dat')
+        <FileInfo for '/tmp/fileinfo_copy.dat': type=FileType.File, size=4>
```

Comment:
Maybe show here that the original file also still exists? (since that is what is different compared to `move`)